common/eth2/src/types.rs

```diff
@@ -1008,15 +1009,40 @@ impl FromStr for Accept {
     type Err = String;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
-        match s {
-            "application/octet-stream" => Ok(Accept::Ssz),
-            "application/json" => Ok(Accept::Json),
-            "*/*" => Ok(Accept::Any),
-            _ => Err("accept header cannot be parsed.".to_string()),
+        let mimes = parse_accept(s);
+
+        // [q-factor weighting]: https://datatracker.ietf.org/doc/html/rfc7231#section-5.3.2
+        let mut highest_mime: Option<Mime> = None;
+        let mut max_pref = 0f32;
+        for m in mimes {
+            let pref = m
+                .get_param("q")
+                .map_or(1f32, |n| n.as_ref().parse::<f32>().unwrap_or(0f32));
+            if pref > max_pref {
+                highest_mime = Some(m);
+                max_pref = pref;
+            }
         }
+
+        highest_mime.map_or(
+            Err("accept header cannot be parsed.".to_string()),
+            |m| match (m.type_(), m.subtype()) {
+                (APPLICATION, OCTET_STREAM) => Ok(Accept::Ssz),
+                (APPLICATION, JSON) => Ok(Accept::Json),
+                (STAR, STAR) => Ok(Accept::Any),
+                _ => Err("accept header is not supported".to_string()),
+            },
+        )
     }
 }
 
+fn parse_accept(accept: &str) -> Vec<Mime> {
+    accept
+        .split(',')
+        .map(|part| part.parse().unwrap())
```

Comment:
We can't use `unwrap` here, or the HTTP handler thread will crash and Lighthouse will return an HTTP 500 internal server error.

This function should be modified to return `Result<Vec<Mime>, String>`. It could use the `collect` instance for `Result<Vec<_>, _>` like this:

```rust
accept.split(',')
    .map(|part| part.parse().map_err(|e| format!("error parsing Accept header: {}", e)))
    .collect()
```